crates/aptos/src/move_tool/mod.rs

```diff
@@ -1418,137 +1404,140 @@ impl FromStr for FunctionArgType {
             "u64" => Ok(FunctionArgType::U64),
             "u128" => Ok(FunctionArgType::U128),
             "u256" => Ok(FunctionArgType::U256),
-            "hex_array" => Ok(FunctionArgType::HexArray),
             "raw" => Ok(FunctionArgType::Raw),
-            str => {
-                // If it's a vector, go one level inside
-                if str.starts_with("vector<") && str.ends_with('>') {
-                    let arg = FunctionArgType::from_str(&str[7..str.len() - 1])?;
-
-                    if arg == FunctionArgType::Raw {
-                        return Err(CliError::CommandArgumentError(
-                            "vector<raw> is not supported".to_string(),
-                        ));
-                    } else if matches!(arg, FunctionArgType::Vector(_)) {
-                        return Err(CliError::CommandArgumentError(
-                            "nested vector<vector<_>> is not supported".to_string(),
-                        ));
-                    } else if arg == FunctionArgType::HexArray {
-                        return Err(CliError::CommandArgumentError(
-                            "nested vector<hex_array> is not supported".to_string(),
-                        ));
-                    }
-
-                    Ok(FunctionArgType::Vector(Box::new(arg)))
-                } else {
-                    Err(CliError::CommandArgumentError(format!("Invalid arg type '{}'.  Must be one of: ['address','bool','hex','hex_array','string','u8','u16','u32','u64','u128','u256','raw', 'vector<inner_type>']", str)))
-                }
-            },
+            str => {Err(CliError::CommandArgumentError(format!(
+                "Invalid arg type '{}'.  Must be one of: ['{}','{}','{}','{}','{}','{}','{}','{}','{}','{}','{}']",
+                str,
+                FunctionArgType::Address,
+                FunctionArgType::Bool,
+                FunctionArgType::Hex,
+                FunctionArgType::String,
+                FunctionArgType::U8,
+                FunctionArgType::U16,
+                FunctionArgType::U32,
+                FunctionArgType::U64,
+                FunctionArgType::U128,
+                FunctionArgType::U256,
+                FunctionArgType::Raw)))
+            }
         }
     }
 }
 
-/// A parseable arg with a type separated by a colon
+#[derive(Parser)]
+pub struct ArgWithTypeVec {
+    /// Arguments combined with their type separated by spaces.
+    ///
+    /// Supported types [address, bool, hex, string, u8, u16, u32, u64, u128, u256, raw]
+    ///
+    /// Vectors may be specified using JSON array literal syntax (you may need to escape this with
+    /// quotes based on your shell interpreter)
+    ///
+    /// Example: `address:0x1 bool:true u8:0 u256:1234 "bool:[true, false]" 'address:[["0xace", "0xbee"], []]'`
```

Comment:
This is pretty clean, might have to make a version bump and note that it's a breaking change in the CHANGELOG.md

Reply:
Can do, I'm glad to add these unless you would rather handle it.